From 94a5c56512cb74fe63b2d65dbec04a86bb9e9b17 Mon Sep 17 00:00:00 2001
From: liuyingjie <yingjie.liu@spacemit.com>
Date: Sun, 4 Jan 2026 20:15:27 +0800
Subject: [PATCH 1/3] backend-drm: set default compositor to gpu

Workaround: spacemit drm has bug to assign multi planes to display.
Force all content to be composited by GPU renderer instead of using
hardware planes.

This changes the default mode from DRM_OUTPUT_PROPOSE_STATE_PLANES_ONLY
to DRM_OUTPUT_PROPOSE_STATE_RENDERER_ONLY in drm_assign_planes().

Patch from: dengbo <bo.deng@spacemit.com>
Date: 2023-12-27
---
 libweston/backend-drm/state-propose.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/backend-drm/state-propose.c b/libweston/backend-drm/state-propose.c
index 1e62e2b..4a908cd 100644
--- a/libweston/backend-drm/state-propose.c
+++ b/libweston/backend-drm/state-propose.c
@@ -973,7 +973,7 @@ drm_assign_planes(struct weston_output *output_base)
 	struct drm_writeback_state *wb_state = output->wb_state;
 	struct weston_paint_node *pnode;
 	struct weston_plane *primary = &output_base->primary_plane;
-	enum drm_output_propose_state_mode mode = DRM_OUTPUT_PROPOSE_STATE_PLANES_ONLY;
+	enum drm_output_propose_state_mode mode = DRM_OUTPUT_PROPOSE_STATE_RENDERER_ONLY;
 
 	assert(output);
 
-- 
2.49.0.windows.1

